src/System.Runtime/tests/System/String.cs

```diff
@@ -88,93 +76,70 @@ public static void TestCtorCharPtrIntInt()
         Assert.Throws<ArgumentOutOfRangeException>(() => new String((char*)null, 5, 1));
     }
 
-    [Fact]
-    public static void TestCtorCharInt()
+    [Theory]
+    [InlineData('a', 0, "")]
+    [InlineData('a', 1, "a")]
+    [InlineData('a', 2, "aa")]
+    [InlineData('a', 3, "aaa")]
+    [InlineData('a', 4, "aaaa")]
+    [InlineData('a', 5, "aaaaa")]
+    [InlineData('a', 6, "aaaaaa")]
+    [InlineData('a', 7, "aaaaaaa")]
+    [InlineData('a', 8, "aaaaaaaa")]
+    [InlineData('a', 9, "aaaaaaaaa")]
+    public static void TestCtorCharInt(char c, int count, string expected)
     {
-        String s;
-
-        // Implementation unrolls copy 4 times.
-
-        s = new String('a', 0);
-        Assert.Equal(String.Empty, s);
-
-        s = new String('a', 1);
-        Assert.Equal("a", s);
-
-        s = new String('a', 2);
-        Assert.Equal("aa", s);
-
-        s = new String('a', 3);
-        Assert.Equal("aaa", s);
-
-        s = new String('a', 4);
-        Assert.Equal("aaaa", s);
-
-        s = new String('a', 5);
-        Assert.Equal("aaaaa", s);
-
-        s = new String('a', 6);
-        Assert.Equal("aaaaaa", s);
-
-        s = new String('a', 7);
-        Assert.Equal("aaaaaaa", s);
-
-        s = new String('a', 8);
-        Assert.Equal("aaaaaaaa", s);
-
-        s = new String('a', 9);
-        Assert.Equal("aaaaaaaaa", s);
-
-        Assert.Throws<ArgumentOutOfRangeException>(() => s = new String('a', -1));
+        String s = new String(c, count);
+        Assert.Equal(expected, s);
     }
 
     [Fact]
-    public static void TestCtorCharArray()
+    public static void TestCtorCharIntInvalid()
     {
-        String s;
-
-        char[] c = { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h' };
-        s = new String(c);
-        Assert.Equal("abcdefgh", s);
-
-        s = new String((char[])null);
-        Assert.Equal(String.Empty, s);
+        Assert.Throws<ArgumentOutOfRangeException>(() => new String('a', -1));
     }
 
-    [Fact]
-    public static void TestCtorCharArrayIntInt()
+    [Theory]
+    [InlineData(new char[] { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h' }, "abcdefgh")]
+    [InlineData(new char[] { 'h', 'e', 'l', 'l', 'o' }, "hello")]
+    [InlineData((char[])null, "")]
```

Comment:
It'd be good to have a few more cases:
- An empty (rather than null) array.
- An array containing a '\0' in the middle.
- An array containing at least one non-ascii character